Sources/MapboxDirections/RouteRefreshing.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+
+/**
+ A skeletal route containing infromation to refresh `Route` object annotations.
+ */
+public protocol RouteRefreshing {
+    var refreshedLegs: [RouteLegRefreshing] { get }
```

Comment:
Can we just call the properties as they are names right now so that conformance is automatic and there's no duplicate "refresh" info in the type and properties?

Reply:
The protocol requires `leg` to be a specific protocol conformance type, so it is problematic to allow both `Route` and `RefreshedRoute` to adopt it at the same time. Also, `RouteLeg.attributes` is internal, so we'll have to add a getter synonym there.